terraform/provider/tls.go

```diff
@@ -3,50 +3,51 @@ package provider
 import (
 	"crypto/tls"
 	"crypto/x509"
-	"errors"
 	"io/ioutil"
 	"os"
 	"time"
 
 	"github.com/gravitational/trace"
 )
 
-var errInvalidCertificate = errors.New("invalid certificate")
-
+// loadTLSConfig loads TLS configuration from given path set
 func loadTLSConfig(certPath, keyPath, rootCAsPath string) (conf *tls.Config, err error) {
```

Comment:
I made a comment above, this can be replaced with `client.LoadKeyPair` once they update to the latest version of `github.com/gravitational/teleport/api`.

Reply:
Done! Removed tls.go as well as dependency on the whole Teleport package completely in favour of new API  package. Thank you for pointing this out!

Reply:
By the way, should these parameters be configurable on Teleport side?

```
	config := client.Config{
		Addrs:           []string{addr},
		Dialer:          nil,
		DialTimeout:     0,
		KeepAlivePeriod: 0,
		KeepAliveCount:  0,
		Credentials:     []client.Credentials{creds},
	}
```

Reply:
By the way, should these parameters be configurable on Teleport side?

```
	config := client.Config{
		Addrs:           []string{addr},
		Dialer:          nil,
		DialTimeout:     0,
		KeepAlivePeriod: 0,
		KeepAliveCount:  0,
		Credentials:     []client.Credentials{creds},
	}
```

Reply:
@Joerger @r0mant Could you please answer?

Reply:
You only need to provide `Credentials` and either `Addrs` or `Dialer`. The other fields will be set to default values if not specified.